lib/util/docker/index.ts

```diff
@@ -0,0 +1,45 @@
+// istanbul ignore file
+
+type Opt<T> = T | null | undefined;
```

Comment:
With strict null checks enabled, it helps to handle notation like this: `{ ..., dockerUser: config.dockerUser, ... }`

Reply:
Yeah, but first we need to convert last js parts to typescript, otherwise we have to add a lot js annotations, which will be removed after conversion.

Feel free to help to convert to typescript.